specification/_types/query_dsl/fulltext.ts

```diff
@@ -271,24 +272,23 @@ export class QueryStringQuery extends QueryBase {
 /**
  * Query flags can be either a single flag or a combination of flags, e.g. `OR|AND|PREFIX`
  * @doc_id supported-flags
- * @codegen_names single, multiple
  */
-export type SimpleQueryStringFlags = SimpleQueryStringFlag | string
```

Comment:
@Anaethelion and me discussed about the semantics and we see multiple options here:

1. Always use explicit array brackets with this marker type in order to avoid changes to the spec compiler. This way even a completely generic generator would generate correct code that allows to pass `0..n` values of this enum.
2. Do not explicitly write out the array brackets, but implicitly assume them. This would make more sense from a semantic perspective as a flags enum value **per definition** has `0..n` bits set, but would either require changes to the spec compiler (to add the array brackets) or to each generator (to make them aware of the `FlagsEnum` concept).

For now, we do like the proposed option 1. better as it requires less code changes and as well does not force the generators to special case the new type alias.

Reply:
@Anaethelion As per discussion with @swallez , we agreed on a third option which looks like this:

```ts
export type SimpleQueryStringFlags = PipeSeparatedFlags<SimpleQueryStringFlag | string>
```

This allows the generator to unwrap the marker interface in order to preserve the current semantics, or to implement custom generation.

Reply:
Actually that would be more

```ts
/**
 * A set of flags that can be represented as a single enum value or a set
 * of values that are encoded as a pipe-separated string
 */
export type PipeSeparatedFlags<T> = T | string

export type SimpleQueryStringFlags = PipeSeparatedFlags<SimpleQueryStringFlag>
```

The rule of thumb is that the TypeScript specification should accept Elasticsearch's JSON out of the box (except for behaviors), and code generators that want to provide more idiomatic types (e.g. an `EnumSet` in Java) can do so by adding specific code generation for the wrapper type.

The name `PipeSeparatedFlags` captures both the semantics (a set of flags) and the encoding, since code generators have to know about the encoding. If another encoding was used elsewhere for a set of flags, we would need another wrapper type.